modin/core/dataframe/pandas/partitioning/partition_manager.py

```diff
@@ -121,7 +127,20 @@ def preprocess_func(cls, map_func):
         `map_func` if the `apply` method of the `PandasDataframePartition` object
         you are using does not require any modification to a given function.
         """
-        return cls._partition_class.preprocess_func(map_func)
+        old_value = PersistentPickle.get()
+        # When performing a function with Modin objects, it is more profitable to
+        # do the conversion to pandas once on the main process than several times
+        # on worker processes. Details: https://github.com/modin-project/modin/pull/6673/files#r1391086755
+        # For Dask, otherwise there may be an error: `coroutine 'Client._gather' was never awaited`
+        need_update = not PersistentPickle.get() and Engine.get() != "Dask"
```

Comment:
@dchigarev I suppose we can leave the current implementation, but for those cases where `to_pandas` is called several times (for example, in different `apply` that go through `preprocess`), enable a mode in which `to_pandas` is called once in the main process.

Reply:
> I suppose we can leave the current implementation

You mean `._inflate_light()`? Where else do we use it? If we are certain that we always want for dataframes to be persistently pickled, then I suppose we shouldn't leave this implementation. There's still a possibility that there are other places in our project where we submit kernels, but don't do this config variables manipulation, which will result into that the `._inflate_light()` implementation will be called against our will.

Reply:
my take is, that we either should drop the `._inflate_light()` implementation at all, or remove these config variable manipulations and always execute `.__reduce__()` in accordance with what a user set in `PersistentPickle` variable. At this point I'm ok with both of the options, it's up to you to decide @anmyachev